spec/The-OpenFF-System.md

```diff
@@ -0,0 +1,185 @@
+
+## Primary components
+
+Components will have discrete responsibilities in order to prevent scope creep and ambiguities about what data can be stored where. Wherever possible, components will have lists of "MUST, MAY, MUST NOT" that specify what data is necessary to construct or convert a component, what a component optionally may know, and what it is forbidden from knowing.
+
+TODO: Populated tables like this for each component below
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| Required things | Optional things | Forbidden things |
+
+### Force field data
+
+System parameters (force field parameters applied to a chemical topology) are represented as the sum of individual components (`PotentialHandler`s). Teach term in a potential energy function is expected to be captured by a `PotentialHandler` or combination thereof. These closely mirror the `ParameterHandler`s in OpenFF Toolkit, and may merge in the future.
+
+Each `PotentialHandler` subclass must specify an string-like `expression` that encodes the algebra of its energy evaluation and a collection of `independent_variables` that specify which variables in the expression do not need to be specified by system parameters. The remaining variables are then expected to be specified in a sequence of `Potential` objects stored in the handler.
+
+```python3
+potential_handler = PotentialHandler(expression="m*x+b",
+independent_variables={"x"}) potential_handler.expression
+```
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| Encode an energy evaluation method | Encode either an algebraic expression or tabluar data |
+| Also encode the necessary data to evaluate the functional form |
+| Import from some common base class
+| Include methods to expose parameter sources
+| Include methods to expose differentiable representations of data
+
+### Chemical topology
+
+The [OpenFF Topology](https://open-forcefield-toolkit.readthedocs.io/en/0.7.2/api/generated/openforcefield.topology.Topology.html#openforcefield.topology.Topology) (or some future form of it) will serve as the representation of the chemical topology. This encodes a graph-like representation of the system _but no physics_. The envisioned scope of this component is roughly equivalent to the data that can be stored in an SDF file.
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| | | |
+| Atoms with elements | Metadata of cheminformatics methods | Force field data |
+| Bonds | Bond constraints
+| | Fractional bond orders
+| Iterators for other valences | Residue/chain/segments
+| Periodicity boolean | Box vectors
```

Comment:
Let's communicate that we're flexible on whether the box vectors could be part of the coordinates.

Reply:
I see them as separate silos of data (them being in the same row here is just a coincidence)